ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -482,65 +485,83 @@ export const WorkspaceEdit = fp.flow(
+              this.props.workspace.billingAccountName
+            ) {
+              // This should never happen but it means the database is out of sync with Google
+              // and does not have the correct billing account stored.
+              // We cannot send over the correct billing account info since the current user
+              // does not have permissions to set it.
+
+              reportError({
+                name: 'Out of date billing account name',
+                message:
+                  `Workspace ${this.props.workspace.namespace} has an out of date billing account name. ` +
+                  `Stored value is ${this.props.workspace.billingAccountName}. ` +
+                  `True value is ${fetchedBillingInfo.billingAccountName}`,
+              });
+            }
+            this.setState({ billingAccountFetched: true });
+          } else {
+            // Otherwise, use this as an opportunity to sync the fetched billing account name from
+            // the source of truth, Google
+            this.setState((prevState) =>
+              fp.set(
+                ['workspace', 'billingAccountName'],
+                fetchedBillingInfo.billingAccountName,
+                prevState
+              )
+            );
+          }
+          this.setState({ billingAccountFetched: true });
+        } catch (e) {
           displayBillingAccounts.push({
             name: this.props.workspace.billingAccountName,
             displayName: 'User Provided Billing Account',
             isFreeTier: false,
             isOpen: true,
           });
-
-          if (
-            fetchedBillingInfo.billingAccountName !==
-            this.props.workspace.billingAccountName
-          ) {
-            // This should never happen but it means the database is out of sync with Google
-            // and does not have the correct billing account stored.
-            // We cannot send over the correct billing account info since the current user
-            // does not have permissions to set it.
-
-            reportError({
-              name: 'Out of date billing account name',
-              message:
-                `Workspace ${this.props.workspace.namespace} has an out of date billing account name. ` +
-                `Stored value is ${this.props.workspace.billingAccountName}. ` +
-                `True value is ${fetchedBillingInfo.billingAccountName}`,
-            });
-          }
-        } else {
-          // Otherwise, use this as an opportunity to sync the fetched billing account name from
-          // the source of truth, Google
           this.setState((prevState) =>
             fp.set(
               ['workspace', 'billingAccountName'],
-              fetchedBillingInfo.billingAccountName,
+              this.props.workspace.billingAccountName,
               prevState
             )
           );
+          this.setState({ fetchBillingAccountError: true });
```

Comment:
I don't know why these setStates were run separately before, but it's good practice to combine these as much as possible, like
```
this.setState({ fetchBillingAccountError: true, billingAccountFetched: false });
```